File: PorraDeportes/src/main/java/es/iespuertodelacruz/yt/porradeportes/entities/Apuesta.java
package es.iespuertodelacruz.yt.porradeportes.entities;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "apuestas")
@NamedQuery(name="Apuesta.findAll", query = "select a from Apuesta a")
public class Apuesta {
    private Integer id;

    private Evento evento;

    private Usuario usuario;

    private String prediccion;

    private BigDecimal cuota;

    private BigDecimal cantidad;

    private String estado;

    public Apuesta(){}

    public Apuesta(Apuesta apuesta){

        this.id = apuesta.getId();
        this.evento = apuesta.getEvento();
        this.usuario = apuesta.getUsuario();
        this.prediccion = apuesta.getPrediccion();
        this.cuota = apuesta.getCuota();
        this.cantidad = apuesta.getCantidad();
        this.estado = apuesta.getEstado();

    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public Apuesta setId(Integer id) {
        this.id = id;
        return this;
    }

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "id_evento", nullable = false)
    public Evento getEvento() {
        return evento;
    }

    public Apuesta setEvento(Evento idEvento) {
        this.evento = idEvento;
        return this;
    }

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "id_usuario", nullable = false)
    public Usuario getUsuario() {
        return usuario;
    }

    public Apuesta setUsuario(Usuario idUsuario) {
        this.usuario = idUsuario;
        return this;
    }

    @Column(name = "prediccion", nullable = false, length = 200)
    public String getPrediccion() {
        return prediccion;
    }


    public Apuesta setPrediccion(String prediccion) {
        this.prediccion = prediccion;
        return this;
    }

    @Column(name = "cuota", nullable = false, precision = 8, scale = 2)
    public BigDecimal getCuota() {
        return cuota;
    }

    public Apuesta setCuota(BigDecimal cuota) {
        this.cuota = cuota;
        return this;
    }

    @Column(name = "cantidad", nullable = false, precision = 8, scale = 2)
    public BigDecimal getCantidad() {
        return cantidad;
    }

    public Apuesta setCantidad(BigDecimal cantidad) {
        this.cantidad = cantidad;
        return this;
    }

    @Column(name = "estado", nullable = false, length = 50)
    public String getEstado() {
        return estado;
    }

    public Apuesta setEstado(String estado) {
        this.estado = estado;
        return this;
    }

}

File: PorraDeportesEloquent/app/Models/Apuesta.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * @property integer $id
 * @property integer $id_evento
 * @property integer $id_usuario
 * @property string $prediccion
 * @property float $cuota
 * @property float $cantidad
 * @property string $estado
 * @property Usuario $usuario
 * @property Evento $evento
 */
class Apuesta extends Model
{
    /**
     * @var array
     */
    protected $fillable = ['id_evento', 'id_usuario', 'prediccion', 'cuota', 'cantidad', 'estado'];

    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function usuario()
    {
        return $this->belongsTo('App\Models\Usuario', 'id_usuario');
    }

    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function evento()
    {
        return $this->belongsTo('App\Models\Evento', 'id_evento');
    }
}
